projects/step-frontend/src/lib/modules/timeseries/chart/time-series-chart.component.ts

```diff
@@ -137,19 +134,17 @@ export class TimeSeriesChartComponent implements OnInit, AfterViewInit, OnChange
       this.legendSettings.zAxisLabel = this.settings.tooltipOptions.zAxisLabel || 'Total';
     }
     settings.series.forEach((series, i) => {
-      if (!series.id) {
-        throw new Error('Every series must have an id');
+      if (series.id) {
+        this.seriesIndexesByIds[series.id] = i + 1; // because the first series is the time
       }
-      this.seriesIndexesByIds[series.id] = i + 1; // because the first series is the time
-
-      series.label = TimeSeriesUtils.mergeSeriesLabels(series.labelItems);
+      series.label = this.mergeLabelItems(series.labelItems);
       this.chartMetadata.push(series.metadata || []);
       if (series.stroke) {
         // aggregate series don't have stroke (e.g total)
         this.legendSettings.items.push({
           seriesId: series.id,
           color: (series.stroke as string) || '#cccccc',
```

Comment:
This was not added in this PR, but I would still move the color to a constant defined in the top of the file like const DEFAULT_STROKE_COLOR = '#ccc';